File: backend/src/main/java/com/iset/covtn/models/Ride.java
package com.iset.covtn.models;

import java.util.Date;
import java.util.List;
import java.util.Objects;

import com.fasterxml.jackson.annotation.*;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor


public class Ride {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private GeoLocation departure;
    @AttributeOverrides({
            @AttributeOverride(name = "latitude", column = @Column(name = "destination_latitude")),
            @AttributeOverride(name = "longitude", column = @Column(name = "destination_longitude"))
    })
    private GeoLocation destination;
    private Date departureTime;
    private Date arrivalTime;

    private boolean approved;

    @ManyToOne
    @NotNull
    private UserInfo driver;

    @OneToMany(fetch = FetchType.LAZY,mappedBy = "ride",cascade = CascadeType.REMOVE)
    private List<RideParticipation> rideParticipations;

    @Override
    public boolean equals(Object o) {
        if (!(o instanceof Ride ride)) return false;
        return Objects.equals(id, ride.id);
    }

    @Override
    public int hashCode() {
        return Objects.hashCode(id);
    }



}

File: mobile/CovTn/app/src/main/java/com/iset/covtn/models/Ride.kt
package com.iset.covtn.models

import android.os.Parcelable
import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey
import kotlinx.parcelize.IgnoredOnParcel
import kotlinx.parcelize.Parcelize
import kotlinx.parcelize.RawValue
import kotlinx.serialization.Transient
import java.util.Date


@Parcelize
data class Ride(
    var id: Long,
    var departure: GeoLocation,
    var destination: GeoLocation,
    var departureTime: Date,
    var arrivalTime: Date,
    var approved: Boolean,
    var driver: User,
    var rideParticipations : List<@RawValue Any>,

): Parcelable{

}
